lib/signaling/v2/peerconnection.js

```diff
@@ -371,7 +380,17 @@ class PeerConnectionV2 extends StateMachine {
       if (!isFirefox) {
         answer = workaroundIssue8329(answer);
       }
-      return this._setLocalDescription(answer);
+      var description = {
+        type: answer.type,
+        sdp: (isChrome || isSafari) && this._isSimulcastRequested()
```

Comment:
@syerrapragada I think this can be its own private method so that you can use it for both local and remote SDPs:
```js
/**
 * @param {RTCSessionDescription} localDescription
 * @param {RTCSessionDescription} localDescriptionWithoutSimulcast
 * @param {RTCSessionDescription} remoteDescription
 * @returns {RTCSessionDescriptionInit}
 */
_maybeApplySimulcastOnlyForVP8MediaSection(
  localDescription,
  localDescriptionWithoutSimulcast,
  remoteDescription) {
  if ((isChrome || isSafari) && this._isSimulcastRequested) {
    return {
      type: localDescription.type,
      sdp: this._revertSimulcastForNonVP8MediaSections(
        localDescription.sdp,
        localDescriptionWithoutSimulcast.sdp,
        remoteDescription.sdp)
    };
  }
  return localDescriptionWithoutSimulcast;
}
```